mne/tests/test_annotations.py

```diff
@@ -403,4 +403,43 @@ def test_annotation_concat():
     assert_equal(len(b), 3)
 
 
+def test_annotations_crop():
+    """test basic functionality of annotation crop."""
+    onset = np.arange(1, 10)
+    duration = np.full_like(onset, 10)
+    description = ["yy"] * onset.shape[0]
+
+    a = Annotations(onset=onset,
+                    duration=duration,
+                    description=description,
+                    orig_time=0)
+
+    # cropping window larger than annotations --> do not modify
+    a_ = a.copy().crop(tmin=0, tmax=42)
+    assert_array_equal(a_.onset, a.onset)
+    assert_array_equal(a_.duration, a.duration)
+
+    # cropping with left shifted window
+    a_ = a.copy().crop(tmin=0, tmax=4.2)
+    assert_array_equal(a_.onset, [1., 2., 3., 4.])
+    # assert_array_equal(a_.duration, [3.2, 2.2, 1.2, 0.2])
+    assert_allclose(a_.duration, [3.2, 2.2, 1.2, 0.2], atol=1e-30)
```

Comment:
I do not understand why array_equal does not work. atol is 1e-30 !!

Reply:
I see the same thing.

```
a_.duration - [3.2, 2.2, 1.2, 0.2]
array([ 0.00000000e+00,  0.00000000e+00, -6.66133815e-16, -7.21644966e-16])
```

so I think that atol=1e-30 is misleading here.